test/testdata/lsp/completion/locals.rb

```diff
@@ -0,0 +1,92 @@
+# typed: true
+extend T::Sig
+
+# This is not defined in any of the methods below.
+# It should never show up in a completion result inside a method.
+x_outside = nil
+
+x_ # error: does not exist
+# ^ completion: x_outside
+
+def without_args
+  x_1 = nil
+  x_ # error: does not exist
+  # ^ completion: x_1
+end
+
+def with_args(x_1)
+  x_2 = nil
+  x_ # error: does not exist
+  # ^ completion: x_1, x_2
+end
+
+# We sort the local variable completion results by name.
+def with_args_backwards(x_2)
+  x_1 = nil
+  x_ # error: does not exist
+  # ^ completion: x_1, x_2
+end
+
+def with_kwargs(x_1, x_2:)
+  x_0 = nil
+  x_ # error: does not exist
+  # ^ completion: x_0, x_1, x_2
+end
+
+# We don't look at which locals are actually in scope (and have no plans to)
+def before_after
+  x_1 = nil
+
+  x_ # error: does not exist
+  # ^ completion: x_1, x_2
+
+  x_2 = 1
+end
+
+# We don't yet complete locals when the position is already a valid local
+def already_valid_local
+  x_1 = nil
+  x_1
+  #  ^ completion: (nothing)
+end
+
+class A
+  def a_method; end
+end
+
+# If there's an explicit reciever, don't suggest similar locals.
+def no_locals_after_dot()
+  a_local = nil
+  A.new.a_ # error: does not exist
```

Comment:
Does it include methods on the parent class or mixins?

Reply:
Yep, there are tests for that in other files, like this one:

https://github.com/sorbet/sorbet/blob/433bf48483761268803d7b086b5e30422be20c19/test/testdata/lsp/completion/class_and_module.rb#L23-L27